src/support/dump-collection.ts

```diff
@@ -141,98 +141,63 @@ class KuzzleDumper extends JSONLDumper {
   }
 }
 
-export async function dumpCollectionData(sdk: any, index: string, collection: string, batchSize: number, destPath: string, query: any = {}, format = 'jsonl') {
+class CSVDumper extends AbstractDumper {
+  constructor(
+    sdk: any,
+    index: string,
+    collection: string,
+    batchSize: number,
+    destPath: string,
+    query: any = {},
+    protected readonly fields: string[]
+  ) {
+    super(sdk, index, collection, batchSize, destPath, query)
+  }
+
+  protected get fileExtension(): string {
+    return 'csv'
+  }
+  setup(): Promise<void> {
+    return new Promise(resolve => resolve())
+  }
+  writeHeader(): Promise<void> {
+    if (this.fields.includes('_id')) {
+      this.fields.splice(this.fields.indexOf('_id'), 1)
+    }
+    this.fields.unshift('_id')
+    return this.writeLine(this.fields.join(','))
+  }
+  writeLine(line: any): Promise<void> {
+    if (!this.writeStream) {
+      throw new Error('Cannot write data: WriteStream is not initialized.')
+    }
+    this.writeStream.write(line)
+    return new Promise(resolve => resolve())
+  }
+  onResult(document: { _id: string; _source: any }): Promise<void> {
+    const values = [document._id, ...pickValues(document._source, this.fields)]
+    return this.writeLine(values)
+  }
+  tearDown(): Promise<void> {
```

Comment:
Same comment about `async` method. Also maybe you could add those empty methods in `AbstractDumper` class and then child class can implements a new version or not